pkg/devenvutil/devenv.go

```diff
@@ -361,6 +363,54 @@ func WaitForAllPodsToBeReady(ctx context.Context, k kubernetes.Interface, log lo
 	return nil
 }
 
+func deleteBackoffPods(ctx context.Context, k kubernetes.Interface, log logrus.FieldLogger, pods []*corev1.Pod) {
+	backoffPods := filter(pods, func(p *corev1.Pod) bool {
+		_, _, isCrashLoopBackoff := debug.ProblemPodCrashLoopBackOff.Detector(ctx, p)
+		return isCrashLoopBackoff
+	})
+
+	if len(backoffPods) == 0 {
+		return
+	}
+	podNames := mapTo(backoffPods, func(p *corev1.Pod) string { return p.Name })
+	log.Infof("Deleting pods to speed things up: %s", strings.Join(podNames, ", "))
+	if err := deletePodsSync(ctx, k, backoffPods); err != nil {
+		log.Warnf("Unable to delelete crashing pods: %s", err)
+	}
+}
+
+func deletePodsSync(ctx context.Context, k kubernetes.Interface, podsToDelete []*corev1.Pod) error {
+	const gracePeriod int64 = 0
+
+	for _, p := range podsToDelete {
+		err := k.CoreV1().Pods(p.Namespace).Delete(ctx, p.Name, *metav1.NewDeleteOptions(gracePeriod))
+		if err != nil {
```

Comment:
nit: When checking error, declare it in-line first like so to avoid `err` shadowing:

```suggestion
		if err := k.CoreV1().Pods(p.Namespace).Delete(ctx, p.Name, *metav1.NewDeleteOptions(gracePeriod)); err != nil {
```